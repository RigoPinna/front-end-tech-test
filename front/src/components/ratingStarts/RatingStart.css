/* .__wrapper_reviews {
  display: flex;
  align-items: center;
  flex-flow: row;

} */
.__wrapper_reviews > p {
  font-size: 12px;
  color: #929292;
}
.rating {
    font-size: 0;
    border: 0;
    text-align: left;
    display: inline-block;
    padding: 5px 10px;
  }
  .rating label {
    margin: 0;
    width: 20px;
    height: 20px;
    display: inline-block;
    position: relative;
    z-index: 3;
  }
  .rating label input {
    height: 0;
    width: 0;
    opacity: 0;
    position: absolute;
  }
  .rating label input + span {
    width: 100%;
    height: 100%;
    position: absolute;
    background-image: url("data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0idXRmLTgiPz4KPCEtLSBHZW5lcmF0b3I6IEFkb2JlIElsbHVzdHJhdG9yIDIxLjAuMCwgU1ZHIEV4cG9ydCBQbHVnLUluIC4gU1ZHIFZlcnNpb246IDYuMDAgQnVpbGQgMCkgIC0tPgo8c3ZnIHZlcnNpb249IjEuMSIgaWQ9IldhcnN0d2FfMSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB4bWxuczp4bGluaz0iaHR0cDovL3d3dy53My5vcmcvMTk5OS94bGluayIgeD0iMHB4IiB5PSIwcHgiCgkgd2lkdGg9IjE4LjkyNnB4IiBoZWlnaHQ9IjE4cHgiIHZpZXdCb3g9IjAgMCAxOC45MjYgMTgiIGVuYWJsZS1iYWNrZ3JvdW5kPSJuZXcgMCAwIDE4LjkyNiAxOCIgeG1sOnNwYWNlPSJwcmVzZXJ2ZSI+CjxnPgoJPHBhdGggZmlsbD0iI0RDQUUzRSIgZD0iTTkuNDYzLDBsMi44OTIsNS45NjhsNi41NywwLjkwOGwtNC43ODIsNC41OTZMMTUuMzExLDE4bC01Ljg0OC0zLjEyOEwzLjYxNCwxOGwxLjE2OC02LjUyOUwwLDYuODc2CgkJbDYuNTctMC45MDhMOS40NjMsMHoiLz4KPC9nPgo8L3N2Zz4=");
    background-size: 100%;
    background-repeat: no-repeat;
    transition: 0.2s;
  }
  .rating label input:checked + span {
    background-image: url("data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0idXRmLTgiPz4KPCEtLSBHZW5lcmF0b3I6IEFkb2JlIElsbHVzdHJhdG9yIDIxLjAuMCwgU1ZHIEV4cG9ydCBQbHVnLUluIC4gU1ZHIFZlcnNpb246IDYuMDAgQnVpbGQgMCkgIC0tPgo8c3ZnIHZlcnNpb249IjEuMSIgaWQ9IldhcnN0d2FfMSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB4bWxuczp4bGluaz0iaHR0cDovL3d3dy53My5vcmcvMTk5OS94bGluayIgeD0iMHB4IiB5PSIwcHgiCgkgd2lkdGg9IjE4LjkyNnB4IiBoZWlnaHQ9IjE4cHgiIHZpZXdCb3g9IjAgMCAxOC45MjYgMTgiIGVuYWJsZS1iYWNrZ3JvdW5kPSJuZXcgMCAwIDE4LjkyNiAxOCIgeG1sOnNwYWNlPSJwcmVzZXJ2ZSI+CjxnPgoJPHBhdGggZmlsbD0iI0QyRDJEMiIgZD0iTTkuNDYzLDQuNTg2bDEuMDkzLDIuMjU1bDAuNDY3LDAuOTYzbDEuMDYsMC4xNDZsMi40ODQsMC4zNDNsLTEuODA4LDEuNzM3bC0wLjc3MSwwLjc0MWwwLjE4OCwxLjA1MwoJCWwwLjQ0MSwyLjQ2N2wtMi4yMS0xLjE4MmwtMC45NDMtMC41MDVMOC41MiwxMy4xMDhsLTIuMjEsMS4xODJsMC40NDEtMi40NjdsMC4xODgtMS4wNTNsLTAuNzcxLTAuNzQxTDQuMzYsOC4yOTJsMi40ODQtMC4zNDMKCQlsMS4wNi0wLjE0Nkw4LjM3LDYuODRMOS40NjMsNC41ODYgTTkuNDYzLDBMNi41Nyw1Ljk2OEwwLDYuODc2bDQuNzgyLDQuNTk2TDMuNjE0LDE4bDUuODQ5LTMuMTI4TDE1LjMxMSwxOGwtMS4xNjctNi41MjkKCQlsNC43ODItNC41OTZsLTYuNTctMC45MDhMOS40NjMsMEw5LjQ2MywweiIvPgo8L2c+CjxnPgoJPHBhdGggZmlsbD0iI0RDQUUzRSIgZD0iTTkuNDYzLDB2MTQuODcyTDMuNjE0LDE4bDEuMTY4LTYuNTI5TDAsNi44NzZsNi41Ny0wLjkwOEw5LjQ2MywweiIvPgo8L2c+Cjwvc3ZnPg==");
  }
  .rating label input:disabled + span {
    background-image: url("data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0idXRmLTgiPz4KPCEtLSBHZW5lcmF0b3I6IEFkb2JlIElsbHVzdHJhdG9yIDIxLjAuMCwgU1ZHIEV4cG9ydCBQbHVnLUluIC4gU1ZHIFZlcnNpb246IDYuMDAgQnVpbGQgMCkgIC0tPgo8c3ZnIHZlcnNpb249IjEuMSIgaWQ9IldhcnN0d2FfMSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB4bWxuczp4bGluaz0iaHR0cDovL3d3dy53My5vcmcvMTk5OS94bGluayIgeD0iMHB4IiB5PSIwcHgiCgkgd2lkdGg9IjE4LjkyNnB4IiBoZWlnaHQ9IjE4cHgiIHZpZXdCb3g9IjAgMCAxOC45MjYgMTgiIGVuYWJsZS1iYWNrZ3JvdW5kPSJuZXcgMCAwIDE4LjkyNiAxOCIgeG1sOnNwYWNlPSJwcmVzZXJ2ZSI+CjxnPgoJPHBhdGggZmlsbD0iI0QyRDJEMiIgZD0iTTkuNDYzLDQuNTg2bDEuMDkzLDIuMjU1bDAuNDY3LDAuOTYzbDEuMDYsMC4xNDZsMi40ODQsMC4zNDNsLTEuODA4LDEuNzM3bC0wLjc3MSwwLjc0MWwwLjE4OCwxLjA1MwoJCWwwLjQ0MSwyLjQ2N2wtMi4yMS0xLjE4MmwtMC45NDMtMC41MDVMOC41MiwxMy4xMDhsLTIuMjEsMS4xODJsMC40NDEtMi40NjdsMC4xODgtMS4wNTNsLTAuNzcxLTAuNzQxTDQuMzYsOC4yOTJsMi40ODQtMC4zNDMKCQlsMS4wNi0wLjE0Nkw4LjM3LDYuODRMOS40NjMsNC41ODYgTTkuNDYzLDBMNi41Nyw1Ljk2OEwwLDYuODc2bDQuNzgyLDQuNTk2TDMuNjE0LDE4bDUuODQ5LTMuMTI4TDE1LjMxMSwxOGwtMS4xNjctNi41MjkKCQlsNC43ODItNC41OTZsLTYuNTctMC45MDhMOS40NjMsMEw5LjQ2MywweiIvPgo8L2c+Cjwvc3ZnPg==") !important;
  }
  .rating label:nth-child(2n) {
    width: 10px;
  }
  .rating label:nth-child(2n) span {
    background-size: 200% !important;
  }
  .rating label:nth-child(2n-1) {
    position: relative;
    margin-left: -10px;
    z-index: 2;
  }
  .rating label:nth-child(2n-1) input:checked + span {
    background-image: url("data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0idXRmLTgiPz4KPCEtLSBHZW5lcmF0b3I6IEFkb2JlIElsbHVzdHJhdG9yIDIxLjAuMCwgU1ZHIEV4cG9ydCBQbHVnLUluIC4gU1ZHIFZlcnNpb246IDYuMDAgQnVpbGQgMCkgIC0tPgo8c3ZnIHZlcnNpb249IjEuMSIgaWQ9IldhcnN0d2FfMSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB4bWxuczp4bGluaz0iaHR0cDovL3d3dy53My5vcmcvMTk5OS94bGluayIgeD0iMHB4IiB5PSIwcHgiCgkgd2lkdGg9IjE4LjkyNnB4IiBoZWlnaHQ9IjE4cHgiIHZpZXdCb3g9IjAgMCAxOC45MjYgMTgiIGVuYWJsZS1iYWNrZ3JvdW5kPSJuZXcgMCAwIDE4LjkyNiAxOCIgeG1sOnNwYWNlPSJwcmVzZXJ2ZSI+CjxnPgoJPHBhdGggZmlsbD0iI0RDQUUzRSIgZD0iTTkuNDYzLDBsMi44OTIsNS45NjhsNi41NywwLjkwOGwtNC43ODIsNC41OTZMMTUuMzExLDE4bC01Ljg0OC0zLjEyOEwzLjYxNCwxOGwxLjE2OC02LjUyOUwwLDYuODc2CgkJbDYuNTctMC45MDhMOS40NjMsMHoiLz4KPC9nPgo8L3N2Zz4=");
  }
  