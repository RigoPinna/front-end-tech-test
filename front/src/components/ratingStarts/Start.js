import React, { useEffect, useState } from 'react'

export const Start = React.memo(({ value, rating }) => {
    const [select, setSelect] = useState( false )
    useEffect(() => {
        (value > rating) 
            && setSelect( true )
    }, [])
    return (
        <label>
            <input disabled={ select } name="star" value={ value } type="radio"/>
            <span></span>
        </label>
    )
})
