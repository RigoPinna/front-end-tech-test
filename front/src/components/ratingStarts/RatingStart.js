import React from 'react'
import './RatingStart.css'
import { Start } from './Start'
const starts =[ 1,1.5,2,2.5,3,3.5,4,4.5,5 ]
export const RatingStart = React.memo(({ _id ,rating, numReviews }) => {
    return (
        <div className='__wrapper_reviews'>
            <fieldset className='rating rating--static'>
            {
                starts.map( start => <Start key={`start-${start}-${_id}`} value={start} rating={rating}/>)
            }
            </fieldset>
            <p>({numReviews} Reviwers)</p>
        </div>
    )
})
