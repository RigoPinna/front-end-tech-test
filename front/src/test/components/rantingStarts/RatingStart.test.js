import React from 'react'
import { mount } from 'enzyme'
import '@testing-library/jest-dom'

import { RatingStart } from '../../../components/ratingStarts/RatingStart'

describe('Pruebas en el componente <RatingStart />',()=> {

    const wrapper = mount(<RatingStart _id={ 1 } rating={3} numReviews={ 12 }/>)
    test('Debe de mostrarse correctamente', () => {
        expect( wrapper).toMatchSnapshot()
    })

    test('Debe de mostrar el nÃºmero de reviews correctamente', () => {
      
        const textReview = wrapper.find('p').text();
        expect( textReview ).toBe("(12 Reviwers)")
    })

    test('Debe de mostrar 0 estrellas activas(Rating)', () => {
        const wrapper = mount(<RatingStart _id={ 1 } rating={0} numReviews={ 12 }/>)
        const starts = wrapper.find('input')
        let selected = starts.map( start => {
            return (start.prop('disabled').valueOf() === false) || false
        });
        selected = selected.filter( sl => sl === true )
        expect(selected.length ).toBe( 0 )
    })

    test('Debe de mostrar 3 estrellas activas(Rating)', () => {
       
        const starts = wrapper.find('input')
        let selected = starts.map( start => {
            return (start.prop('disabled').valueOf() === false) || false
        });
        selected = selected.filter( sl => sl === true )
        expect(selected.length ).toBe( 5 )
    })

})