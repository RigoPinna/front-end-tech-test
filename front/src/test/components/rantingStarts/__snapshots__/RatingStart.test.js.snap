// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <RatingStart /> Debe de mostrarse correctamente 1`] = `
<div
  className="__wrapper_reviews"
>
  <fieldset
    className="rating rating--static"
  >
    <label>
      <input
        disabled={false}
        name="star"
        type="radio"
        value={1}
      />
      <span />
    </label>
    <label>
      <input
        disabled={false}
        name="star"
        type="radio"
        value={1.5}
      />
      <span />
    </label>
    <label>
      <input
        disabled={false}
        name="star"
        type="radio"
        value={2}
      />
      <span />
    </label>
    <label>
      <input
        disabled={false}
        name="star"
        type="radio"
        value={2.5}
      />
      <span />
    </label>
    <label>
      <input
        disabled={false}
        name="star"
        type="radio"
        value={3}
      />
      <span />
    </label>
    <label>
      <input
        disabled={true}
        name="star"
        type="radio"
        value={3.5}
      />
      <span />
    </label>
    <label>
      <input
        disabled={true}
        name="star"
        type="radio"
        value={4}
      />
      <span />
    </label>
    <label>
      <input
        disabled={true}
        name="star"
        type="radio"
        value={4.5}
      />
      <span />
    </label>
    <label>
      <input
        disabled={true}
        name="star"
        type="radio"
        value={5}
      />
      <span />
    </label>
  </fieldset>
  <p>
    (
    12
     Reviwers)
  </p>
</div>
`;
